extensions/warp-blink-wrtc/src/rtp_logger.rs

```diff
@@ -0,0 +1,147 @@
+// this module is purposely not included in lib.rs. it might be needed later.
+
+use anyhow::{bail, Result};
+use once_cell::sync::Lazy;
+use std::{
+    fs::{self, create_dir_all},
+    io::{BufWriter, Write},
+    path::PathBuf,
+    sync::atomic::{AtomicBool, Ordering},
+};
+use tokio::sync::mpsc::{Receiver, Sender};
+use uuid::Uuid;
+use warp::sync::{Arc, RwLock};
+use webrtc::rtp::header::Header;
+
+pub struct LoggerInstance {
+    peer_id: String,
+    tx: Sender<RtpHeaderWrapper>,
+}
+
+pub struct RtpHeaderWrapper {
+    val: Header,
+    id: String,
+}
+
+impl LoggerInstance {
+    pub fn log(&self, header: Header) {
+        let _ = self.tx.try_send(RtpHeaderWrapper {
+            val: header,
+            id: self.peer_id.clone(),
+        });
+    }
+}
+
+struct RtpLogger {
+    tx: Sender<RtpHeaderWrapper>,
+    should_quit: Arc<AtomicBool>,
+    should_log: bool,
+}
+
+static RTP_LOGGER: Lazy<RwLock<Option<RtpLogger>>> =
+    once_cell::sync::Lazy::new(|| RwLock::new(None));
+
+pub async fn init(call_id: Uuid, log_path: PathBuf) -> Result<()> {
+    deinit().await;
+
+    if !log_path.exists() {
+        if let Err(e) = create_dir_all(&log_path) {
+            log::error!("failed to create directory for rtp_logger: {e}");
+            bail!(e);
+        }
+    }
+
+    let (tx, rx) = tokio::sync::mpsc::channel(1024 * 5);
+    let should_quit = Arc::new(AtomicBool::new(false));
+
+    let logger = RtpLogger {
+        tx,
+        should_quit: should_quit.clone(),
+        should_log: true,
+    };
+    RTP_LOGGER.write().replace(logger);
+
+    std::thread::spawn(move || {
```

Comment:
I think a call can last long enough that in theory it may be preferable to use a thread. I say "in theory" because I don't think Uplink would exhaust the thread pool available to `spawn_blocking`.